src/dnx.clr.managed/DomainManager.cs

```diff
@@ -57,7 +69,50 @@ private int Main(int argc, string[] argv)
         },
         null);
 
-        return RuntimeBootstrapper.Execute(argv);
+        return RuntimeBootstrapper.Execute(argv, _dnxTfm);
+    }
+
+    private Version DetermineDnxVersion(string applicationBase)
+    {
+        var lockFilePath = Path.Combine(applicationBase, "project.json");
+        try
+        {
+            // Parse the project
+            JsonObject json;
+            using (var reader = File.OpenText(lockFilePath))
+            {
+                json = JsonDeserializer.Deserialize(reader) as JsonObject;
+                if (json == null)
+                {
+                    Logger.TraceError($"[{nameof(DomainManager)}] project.json did not contain a JSON object at the root.");
+                    return DefaultVersion;
+                }
+            }
+
+            // Find the highest DNX desktop version, if any and map it to .NET version
+            var version = (json.ValueAsJsonObject("frameworks")?.Keys ?? Enumerable.Empty<string>())
+                .Select(k =>
+                {
+                    FrameworkName fx;
+                    if (!dnx.host.FrameworkNameUtility.TryParseFrameworkName(k, out fx))
+                    {
+                        return null;
+                    }
+                    return fx;
+                })
+                .Where(fx => string.Equals(fx?.Identifier, FrameworkNames.LongNames.Dnx))
+                .Max(fx => fx.Version);
```

Comment:
Seems this exception can be handled by the catch block below, but I guess the log there doesn't capture the empty sequence case.